oshi-core/src/main/java/oshi/util/ParseUtil.java

```diff
@@ -489,4 +493,31 @@ public static int getNthIntValue(String line, int n) {
         }
         return 0;
     }
+
+    /**
+     * Removes all sub strings from the string
+     * @param original source String to remove from
+     * @param toRemove the sub string to be removed
+     * @return
+     */
+    public static String removeMatchingString(final String original, final String toRemove) {

```

Comment:
After thinking about it, I think I'm fond of your first, simplest, `removeFirstMatchingString()` for this specific application.

For the bigger picture, I'd like a more general key/value parsing method that handles all these specific cases with options to handle how to find the key, delimiter, whitespace, quotes around value, etc.